compiler/rustc_hir_analysis/src/check/entry.rs

```diff
@@ -124,6 +124,10 @@ fn check_main_fn_ty(tcx: TyCtxt<'_>, main_def_id: DefId) {
     if let Some(term_did) = tcx.lang_items().termination() {
         let return_ty = main_fnsig.output();
         let return_ty_span = main_fn_return_type_span(tcx, main_def_id).unwrap_or(main_span);
+        if return_ty.has_bound_regions() {
```

Comment:
Can you use [`no_bound_vars`](https://doc.rust-lang.org/stable/nightly-rustc/rustc_middle/ty/struct.Binder.html#method.no_bound_vars) instead?